enhancements/network/persistent-ips-sdn-secondary-networks.md

```diff
@@ -807,3 +807,99 @@ allocations, it is probable the time it takes to provision a pod to grow.
+in the interface's `NetworkSelectionElements`.
+
+This flow is portrayed in the following sequence diagram:
+
+```mermaid
+sequenceDiagram
+    actor user
+    user ->> kubevirt: Create(vm)
+    note over kubevirt, apiserver: leaseName = <vm>.<network>.<interface>
+    kubevirt->>apiserver: Create(ipamLease{name=leaseName})
+    apiserver->>ipamLeaseOperator: Reconcile(ipamLease)
+    Note right of ipamLeaseOperator: Status has no IPs
+    ipamLeaseOperator ->> ipamLeaseOperator: subnet = GetNAD(ipamLease).Subnet
+    ipamLeaseOperator ->> ipamLeaseOperator: ips = AllocateNextIPs(subnet)
+    ipamLeaseOperator ->> apiserver: UpdateStatus(ipamLease, ips)
+    loop until status has IPs
+        kubevirt->>apiserver: Get(ipamLease)
+    end
+    kubevirt ->> apiserver: Create(virtLauncher{ips: ipamLeases.IPs})
+```
+
+The flow for **stopping** the VM is quite simple - and similar to what we've
+seen so far; the user issues a VM stop command, which will trigger the launcher
+pod deletion. The persistent IP will **not** be deleted.
+
+```mermaid
+sequenceDiagram
+    actor user
+
+    user ->> kubevirt: Stop(vm)
+    kubevirt-->>user: Result
+    kubevirt->>apiserver: Delete(virtLauncher)
+```
+
+The flow for starting a stopped VM is again quite familiar; on a VM start event
+KubeVirt realizes there's already a persistent IP allocation for its interface,
+and will template the pod with it.
+
+```mermaid
+sequenceDiagram
+    actor user
+
+    user ->> kubevirt: Start(vm)
+    kubevirt->>apiserver: Get(ipamLease{name=vm.network.interface})
+    apiserver-->>kubevirt: ipamLease
+    kubevirt ->> apiserver: Create(virtLauncher{ips: ipamLeases.Status.IPs})
+```
+
+The flow for deleting a VM is similar to what we've seen so far; the Kubernetes
+garbage collector is the entity responsible for deleting the persistent IP
+allocation.
+
+```mermaid
+sequenceDiagram
+    actor user
+
+    user ->> apiserver: Delete(vm)
+    apiserver ->> gc: Delete(vm)
+    gc->>apiserver: Delete(virtLauncherPod)
+    apiserver-->>gc: Result
+    gc->>apiserver: Delete(ipamLease)
+    apiserver-->>gc: Result
+    apiserver->>ipamLeaseOperator: Reconcile(ipamLease)
+    ipamLeaseOperator ->> ipamLeaseOperator: DeallocateIPs(ipamLease.Status.IPs)
+```
+
+#### Advantages of this alternative
+
+The OVN-Kubernetes existing IPAM solution would only require a minor tweak: we
+would need to allow static IP requests on NADs with a defined subnet attribute.
+
+This solution would be generic: other workloads interested in this feature
+would just have to create the CR, and wait until an IP is reported in its
+status, then template the pod accordingly.
+
+#### Disadvantages of this alternative
+
+- KubeVirt would have to "track" a new CRD.
+- KubeVirt memory footprint would increase
+- Reinventing the IPAM wheel
+- Productifying a new IPAM solution takes time
+- We're basically implementing IPAM outside CNI. Makes a clunky flow.
```

Comment:
Also making kubevirt wait for ip at status is not very k8s async way of things.

Reply:
Why not ? AFAIU, that's literally the definition of a control loop. 

FWIW this is not an active wait. We would re-queue the operation, and  retry using exponential backoff. Only once the status IP is reported would the operation be fulfilled.